### DISCUSSION TITLE
Limit power consumption rule possible?

### DISCUSSION THREAD

**Discussion by conkydong** [arendst/Tasmota#21396] [General]
Would it somehow be possible to limit the power consumption per hour/day/week via rules? If so, how?

**Comment by sfromis** [comment]
If you mean of the Tasmota device itself (as opposed to what may be controlled by Tasmota), rules are not really involved in that. What consumes variable power is mostly Wifi activity. The main avenue for cutting down is using deep sleep, with the device waking up periodically. Not suitable for short "pauses". The value of `Sleep` can influence power consumption, but normally not much compared to Wifi.

**Reply by conkydong** [reply]
I'm sorry, I might not have made myself clear. I meant whether it is possible to block an output (multiple socket or entire appliance) when a certain amount of power has been consumed.

For example: A small heater has consumed 3 kWh and the output should then remain blocked for a day and only then be unblocked.

Is this possible somehow?

**Reply by sfromis** [reply]
Well, blocking depends on what means may be used to power it on, and 3 KWh since when. If it is by on-device button, you could decouple the input from the relay action, and have rules deciding when to turn it on/off after all, rules to save current energy consumed level at one time, and add the max, where you then could have another rule to detect energy consumption exceeding the max, power off and set a variable for blocking, to be used by the input detection rule. And of course also a way of deciding when to unblock, like daily or whatever.

**Reply by arendst** [reply]
There are standard energy features regarding max power used during a day. Please look into the energy driver for long standing support for this on request from African energy suppliers.

**Reply by arendst** [reply]
Checkout commands ``MaxPower``, ``MaxPowerHold``, ``MaxPowerWindo``, ``MaxPowerReached``, ``MaxPowerReachedRetry``, ``SafePower``, ``SafePowerHold``, ``SafePowerWindow``, ``PowerMonitor``, ``MaxEnergy`` and ``MaxEnergyStart``.
